test/functional/wallet-encryption.py

```diff
@@ -56,6 +56,21 @@ def run_test(self):
         assert_raises_rpc_error(-14, "wallet passphrase entered was incorrect", self.nodes[0].walletpassphrase, passphrase, 10)
         self.nodes[0].walletpassphrase(passphrase2, 10)
         assert_equal(privkey, self.nodes[0].dumpprivkey(address))
+        self.nodes[0].walletlock()
+
+        # Test timeout bounds
+        assert_raises_rpc_error(-8, "Timeout cannot be negative.", self.nodes[0].walletpassphrase, passphrase2, -10)
+        # Check the timeout
+        # Check a time less than the limit
+        expected_time = int(time.time()) + (1 << 31) - 600
+        self.nodes[0].walletpassphrase(passphrase2, (1 << 31) - 600)
+        actual_time = self.nodes[0].getwalletinfo()['unlocked_until']
+        assert expected_time <= actual_time < expected_time + 5 # 5 second buffer
```

Comment:
Might want to use the provided `assert_greater...` helper methods to give a more verbose output in case of failure.

Current error shows only:

```
File "/home/travis/build/bitcoin/bitcoin/build/bitcoin-i686-pc-linux-gnu/test/functional/wallet-encryption.py", line 68, in run_test
    assert expected_time <= actual_time < expected_time + 5 # 5 second buffer
AssertionError
```

and not by how much it was off.

Reply:
Done.